R/sapflow-Rs.Rmd

```diff
@@ -218,24 +218,59 @@ ggplot(wx_dat, aes(DOY, PAR)) +
   geom_point(aes(y = Daytime_PAR, color = PAR_group))
 ```
 
+```{r sapwood-area-calc, echo = FALSE}
+
+inventory %>% 
+  filter(Plot == "Control", Species_code != "FAGR") %>% 
+  drop_na(Sapflux) %>% 
+  select(Sapflux, DBH_2019, Species_code) %>% 
+  rename(Tree = Sapflux, DBH = DBH_2019, Species = Species_code) -> js_trees
+
+# To calculate sapflux, sapwood area needs to be calculated from sapwood depth
+sapwood_depth <- read_csv("~/Desktop/sapwood_depthsSERC.csv", 
+                          col_names = c("DBH", "sapwood_d", "Species"), 
+                          skip = 1) # read in randomly sampled sapwood depths at site
+
+# A linear model is used to predict our tree's sapwood area
+sd_model <- lm(sapwood_d ~ DBH * Species, data = sapwood_depth)
+
+js_trees$pred_sd <- predict.lm(object = sd_model, newdata = js_trees)
+
+# Test to make sure predictor works 
+ggplot(sapwood_depth, aes(x = DBH, y = sapwood_d, colour = Species)) + 
+  geom_point() + 
+  geom_smooth(method = lm, se =  FALSE) + 
+  geom_point(data = js_trees, aes(x = DBH, y = pred_sd, 
+                                  colour = Species), shape = 1, size = 5)
+
+js_trees %>% 
+  # calculate area of the tree and heartwood, the difference will give us sapwood area
+  mutate(t_area_cm2 = pi * (DBH / 2)^2, hw_area_cm2 = pi * ((DBH - 2 * pred_sd) / 2) ^ 2) %>% 
+  mutate(sw_area_cm2 = t_area_cm2 - hw_area_cm2) -> sapwood_area
+
+```
 
 ```{r data-cleaning, echo = FALSE, message = FALSE}
 
 # First, we need to filter for the Js trees that also have Rs measurements
 rs_trees <- unique(ports_lt$Tree)
 
 sapflow_k_long %>% 
+  left_join(sapwood_area, by = "Tree") %>% 
   # Then, we need to transform K to Js (sapflux density) through a conversion
-  mutate(Js = 119* 10^(-6) * K^(1.231)) %>% 
+  mutate(Js = 119* 10^(-6) * K^(1.231)) %>%
```

Comment:
Where do the 119 and 1.231 come from? Is there a citation we can put here?

Reply:
This is from Granier 1987, 10.1093/treephys/3.4.309. I've added this in the comments